test/common/network/connection_impl_test.cc

```diff
@@ -2950,6 +2950,58 @@ TEST_F(PipeClientConnectionImplTest, SkipSourceAddress) {
   connection->close(ConnectionCloseType::NoFlush);
 }
 
+// Validate the rtt for the connection socket
+TEST_P(ConnectionImplTest, RttBySocket) {
```

Comment:
WDYT about using `IoSocketHandleImpl::lastRoundTripTime` in this test to make the test cross platform?

Reply:
I'm neutral about optref vs other types but if it is easy to make the test work cross-platform I think it's worth doing